apps/examples/connect_test/connect_test_client.c

```diff
@@ -0,0 +1,676 @@
+static int _udp_send_data(int fd, int size, struct sockaddr_in* saddr)
+{
+	char buf[WT_BUF_SIZE];
+	int remain = size;
+	int slen = sizeof(struct sockaddr_in);
+	int cnt = 1;
+	int total_sent = 0; 
+	CT_LOG(TAG, "_udp_send_data start");
+	while (remain > 0) {
+		int send_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+		int nbytes = sendto(fd, buf, send_size, 0, (struct sockaddr*)saddr, slen);
+		if (nbytes == 0) {
+			CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+			CT_LOGE(TAG, "connection closed");
+			return -1;
+		} else if (nbytes < 0) {
+			if (errno == EWOULDBLOCK) {						 
+				CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+				CT_LOGE(TAG, "timeout error %d", errno);
+				return -1;
+			} else {
+				CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+				CT_LOGE(TAG, "connection error %d", errno);
+				return -1;
+			}
+		}
+		total_sent += nbytes;
+		remain -= nbytes;
+		cnt++;
+	}
+	CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+	return 0;
+}
+
+static int _udp_recv_data(int fd, int size, struct sockaddr_in* saddr)
+{
+	char buf[WT_BUF_SIZE];
+	unsigned int slen = sizeof(struct sockaddr_in);
+	int remain = size;
+	int cnt = 1;
+	int total_received = 0;
+	CT_LOG(TAG, "_udp_recv_data start");
+	while (remain > 0) {
+		int read_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+		int nbytes = recvfrom(fd, buf, read_size, 0, (struct sockaddr*)saddr, &slen);
+		if (nbytes == 0) {
+			CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+			CT_LOGE(TAG, "connection closed");
+			return -1;
+		} else if (nbytes < 0) {
+			if (errno == EWOULDBLOCK) {
+				CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+				CT_LOGE(TAG, "timeout error %d", errno);
+				return -1;
+			} else {
+				CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+				CT_LOGE(TAG, "connection error %d", errno);
+				return -1;
+			}
+		}
+		total_received += nbytes;
+		remain -= nbytes;
+		cnt++;
+	}
+	CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+	return 0;
+}
+
+static int _udp_client(int size)
+{
+	struct sockaddr_in saddr;
+	int ret = 0;
+	int sockfd = 0;
+	
+	/* Socket Creation */
+	CT_LOG(TAG, "create socket");
+	sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sockfd < 0) {
+		CT_LOGE(TAG, "create socket fail %d", errno);
+		return -1;
+	}
+	
+	/* Set Timeout for Sockets data transfer */
+	struct timeval tv;
+	tv.tv_sec = 30;
+	tv.tv_usec = 0;
+	ret = setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv));
+
+	/* Connect the socket to the server */
+	CT_LOG(TAG, "connect to %s:%d", WT_UDP_SERVER_ADDR, WT_UDP_PORT);
+	saddr.sin_family = AF_INET;
+	saddr.sin_port = htons(WT_UDP_PORT);
+	inet_aton(WT_UDP_SERVER_ADDR, &(saddr.sin_addr));
+	
+	unsigned int clen = sizeof(struct sockaddr_in);
+	int slen = sizeof(struct sockaddr_in);
+	CT_LOG(TAG, "Sending Device MAC to peer, mac: %s", g_mac_buf);
+	int nbytes = sendto(sockfd, g_mac_buf, sizeof(g_mac_buf), 0, (struct sockaddr*)&saddr, slen);
+	if (nbytes == 0) {
```

Comment:
How about to make new functions for sendto and recvfrom. Both API needs exception handling so it's difficult to read. If you make it separate function, readability will increase.

Reply:
Changes done as suggested